src/StyledNestableList/StyledNestableList.js

```diff
@@ -0,0 +1,182 @@
+      </Box>
+    );
+  };
+
+  renderPrefix = data => {
+    if (
+      data.isPreview ||
+      isLastItem(data.siblings, data.item) ||
+      isRootItem(data.depth)
+    ) {
+      return null;
+    }
+    return <div className={classes.childArrow} />;
+  };
+
+  renderItem = options => {
+    const { isPlaceholder, depth, isPreview, siblings, item } = options;
+    const { id, children, isCollapsed, ...rest } = item;
+    const isLastChild = isLastItem(siblings, item);
+    return (
+      <Box>
+        {!isRootItem(depth) &&
+          this.renderArrow({ isLastChild, isPreview, depth })}
+        <Box
+          display={'block'}
+          className={classNames(
+            {
+              [classes.rootItem]: isRootItem(depth),
+              [classes.firstItem]: isFistItem(siblings, item),
+              [classes.lastItem]: isLastItem(siblings, item),
+              [classes.haveChildren]: Boolean(item.children),
+              [classes.draggablePlaceholder]: isPlaceholder,
+              [classes.draggablePreview]: isPreview,
+            },
+            classes.item,
+          )}
+          width={'100%'}
+        >
+          <TableListItem
+            {...rest}
+            showDivider={false}
+            options={rest.options ? rest.options : []}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
+  render() {
+    const {
+      dataHook,
+      className,
+      items,
+      maxDepth,
+      onChange,
+      preventChangeDepth,
+    } = this.props;
+
+    return (
+      <div data-hook={dataHook} className={st(classes.root, {}, className)}>
+        <NestableList
+          items={items}
+          renderItem={this.renderItem}
+          renderPrefix={this.renderPrefix}
+          renderAction={this.renderAction}
+          onUpdate={onChange}
+          maxDepth={maxDepth}
+          threshold={30}
+          theme={theme}
+          preventChangeDepth={preventChangeDepth}
+          childrenStyle={{
+            position: 'relative',
+            marginLeft: '84px',
+          }}
+          childrenProperty={'children'}
+          isRenderDraggingChildren
+        />
+      </div>
+    );
+  }
+}
+
+StyledNestableList.displayName = 'StyledNestableList';
+
+const Node = {
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  isCollapsed: PropTypes.bool,
+  ...TableListItem.propTypes,
+};
+
+const NodeShape = PropTypes.shape(Node);
+Node.children = PropTypes.arrayOf(NodeShape);
+
+StyledNestableList.propTypes = {
+  /** label for add new item button */
+  addItemLabel: PropTypes.string,
+  /** triggers in case when add new item button is clicked */
+  onAddItem: PropTypes.func,
+  /** triggers when order or nesting is changed */
+  onChange: PropTypes.func,
+  /** Maximum depth of the node three */
+  maxDepth: PropTypes.number,
+  /** Allow drop node only on it's own depth */
+  preventChangeDepth: PropTypes.bool,
+  /** Nodes to render */
+  items: PropTypes.arrayOf(NodeShape),
```

Comment:
Unfortunately our AutoDocs doesn't know how to display complex types, so just write it down as a comment
```suggestion
  /**
   * * __id__ - Specifies the item id
   * * __isCollapsed__ - Whether to render the items children
   * * __children__ - A list of children of type item
   * * All other TableListItem props
   */
  items: PropTypes.arrayOf(NodeShape),
```